### PR TITLE
:bug: ActivityFlags no longer crash

### PR DESCRIPTION
**Pull request by Lunarmagpie**
I got so annoyed that I finally fixed it
ActivityFlags are sent as a number with multiple flags in it. Pretty sure I did the bitwise to get the bool right.

 ### Check off the following

-   [x] I have tested my changes with the current requirements
-   [x] My Code follows the pep8 code style.

### CODE REVIEW

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pincer/objects/events/presence.py [hunk 1]
```diff
@@ -132,13 +131,32 @@ class ActivitySecrets(APIObject):
...
 
 
-class ActivityFlags(IntEnum):
-    INSTANCE = 1 << 0
-    JOIN = 1 << 1
-    SPECTATE = 1 << 2
-    JOIN_REQUEST = 1 << 3
-    SYNC = 1 << 4
-    PLAY = 1 << 5
+@dataclass
```
[the remark this answers is not in the document]
Comment (Lunarmagpie) [reply pincer/objects/events/presence.py:134]: The free `repr` is worth keeping the dataclass for

**Review by Endercheif — APPROVED** [review]

File: pincer/objects/events/presence.py [hunk 2]
```diff
@@ -132,13 +131,32 @@ class ActivitySecrets(APIObject):
...
+    """
+    INSTANCE: bool
+    JOIN: bool
+    SPECTATE: bool
+    JOIN_REQUEST: bool
+    SYNC: bool
+    PLAY: bool
+    PARTY_PRIVACY_FRIENDS: bool
+    PARTY_PRIVACY_VOICE_CHANNEL: bool
+    EMBEDDED: bool
```
Comment (trag1c) [inline pincer/objects/events/presence.py:148]: Don't think this is needed if there's an overriding `__init__`

**Comment by Lunarmagpie** [comment]
Maybe i should remove the dataclass :skull: 
i think the usage was unpythonic anyway

**Comment by Lunarmagpie** [comment]
dataclass is gone

**Pull request merged**